File: src/Services/Totem.Application/Services/ServiceTypeServices/IServiceTypeService.cs
using Totem.Domain.Models.ServiceTypeModels;

namespace Totem.Application.Services.ServiceTypeServices
{
	public interface IServiceTypeService
	{
		Task<(Result result, ServiceTypeView data)> GetByIdAsync(Guid id);
		Task<(Result result, List<ServiceTypeSummary> data)> GetActiveServicesAsync();
		Task<(Result result, List<ServiceTypeSummary> data)> GetAllAsync();
		Task<(Result result, Guid data)> CreateAsync(ServiceTypeRequest request);
		Task<Result> UpdateAsync(Guid id, ServiceTypeRequest request);
		Task<Result> ToggleStatusAsync(Guid id);
		Task<Result> DisableAsync(Guid id);
	}
}

File: src/Web/src/services/interfaces/IServiceTypeService.ts
import type { ServiceResult } from "../../models/baseServiceModels";
import type { ServiceTypeRequest, ServiceTypeSummary, ServiceTypeView } from "../../models/ServiceTypeModels";

export interface IServiceTypeService {
  getByIdAsync(id: string): Promise<ServiceResult<ServiceTypeView>>;
  getActiveServicesAsync(): Promise<ServiceResult<ServiceTypeSummary[]>>;
  getAllAsync(): Promise<ServiceResult<ServiceTypeSummary[]>>;
  createAsync(request: ServiceTypeRequest): Promise<ServiceResult<string>>;
  updateAsync(id: string, request: ServiceTypeRequest): Promise<ServiceResult<void>>;
  toggleStatusAsync(id: string): Promise<ServiceResult<void>>;
  disableAsync(id: string): Promise<ServiceResult<void>>;
}
